pilot/pkg/model/push_context.go

```diff
@@ -1627,32 +1630,31 @@ func (ps *PushContext) initDefaultExportMaps() {
 // with the proxy and derive listeners/routes/clusters based on the sidecar
 // scope.
 func (ps *PushContext) initSidecarScopes(env *Environment) {
-	sidecarConfigs := env.List(gvk.Sidecar, NamespaceAll)
+	rawSidecarConfigs := env.List(gvk.Sidecar, NamespaceAll)
 
-	sortConfigByCreationTime(sidecarConfigs)
+	sortConfigByCreationTime(rawSidecarConfigs)
 
-	sidecarConfigWithSelector := make([]config.Config, 0)
-	sidecarConfigWithoutSelector := make([]config.Config, 0)
-	for _, sidecarConfig := range sidecarConfigs {
+	sidecarConfigs := make([]config.Config, 0, len(rawSidecarConfigs))
+	for _, sidecarConfig := range rawSidecarConfigs {
```

Comment:
only reason it was not done like this earlier to avoid looping twice. But since sidecars wont be that many - this should be fine I guess

Reply:
Looping twice is cheap, but splitting and merging introduces unnecessary overhead. In addition, the number of sidecars is not always small, for example, there are 4000+ sidecar CRs in our production.